sky/clouds/ibm.py

```diff
@@ -0,0 +1,454 @@
+        requirements.  Currently, this function implements "filtering" the
+        cloud's offerings only w.r.t. accelerators constraints.
+
+        Launchable resources require a cloud and an instance type be assigned.
+        """
+        fuzzy_candidate_list = []
+        if resources.instance_type is not None:
+            assert resources.is_launchable(), resources
+            resources = resources.copy(accelerators=None)
+            return ([resources], fuzzy_candidate_list)
+
+        def _make(instance_list):
+            resource_list = []
+            for instance_type in instance_list:
+                r = resources.copy(
+                    cloud=IBM(),
+                    instance_type=instance_type,
+                    # Setting this to None as IBM doesn't separately bill /
+                    # attach the accelerators.  Billed as part of the VM type.
+                    accelerators=None,
+                )
+                resource_list.append(r)
+            return resource_list
+
+        # Currently, handle a filter on accelerators only.
+        accelerators = resources.accelerators
+        if accelerators is None:
+            # No requirements to filter, so just return a default VM type.
+            return (_make([IBM.get_default_instance_type()]),
+                    fuzzy_candidate_list)
+
+        assert len(accelerators) == 1, resources
+        acc, acc_count = list(accelerators.items())[0]
+        (instance_list, fuzzy_candidate_list
+        ) = service_catalog.get_instance_type_for_accelerator(acc,
+                                                              acc_count,
+                                                              clouds='ibm')
+        if instance_list is None:
+            return ([], fuzzy_candidate_list)
+        return (_make(instance_list), fuzzy_candidate_list)
+
+    @classmethod
+    def _get_default_region(cls) -> clouds.Region:
+        region_name = get_cred_file_field('region', 'us_south')
+
+        return next(
+            region for region in cls.regions() if region.name == region_name)
+
+    @classmethod
+    def get_default_image(cls, region):
+        """
+        Returns default image.
+        if user ran the ibm script to creating a gpu image,
+        its id is loaded from local cache,
+        otherwise provides a clean image of ubuntu 22-04.
+        """
+
+        def _get_image_objects():
+            # pylint: disable=E1136
+            images = []
+            res = client.list_images().get_result()
+            images.extend(res['images'])
+
+            while res.get('next'):
+                link_to_next = res['next']['href'].split('start=')[1].split(
+                    '&limit')[0]
+                res = client.list_images(start=link_to_next).get_result()
+                images.extend(res['images'])
+            return images
+
+        # if user specified an image id for the region, return it.
```

Comment:
Can't use a single image for all region, since images are region locked(region-specific).
I'm currently allowing users to specify `image_id:region:ID` in `.ibm/credentials.yaml`, and of course, I'll try to implement it via `--image_id`, just not sure it will be added to this PR.

Reply:
Great, agreed.